org.mavlink.qgroundcontrol.yml

```
@@ -0,0 +1,43 @@
+app-id: org.mavlink.qgroundcontrol
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+command: QGroundControl
+rename-appdata-file: org.mavlink.qgroundcontrol.metainfo.xml
+
+finish-args:
+- --socket=wayland
+- --socket=fallback-x11
+- --socket=pulseaudio
+- --device=dri
+- --share=ipc
+- --share=network
+
+cleanup:
+- /include
+- /bin/dbf*
+- /bin/shp*
+- /lib/cmake
+- /share/shapelib
+- "*.a"
+- "*.cmake"
+
+modules:
+  - name: QGroundControl
+    sources:
+    - type: git
+      url: https://github.com/mavlink/qgroundcontrol.git
+      commit: eca2614b6e2a237c3717f246c8c6e42d7d903aaf # Mar 1, 2023
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+    - -D BUILD_SHARED_LIBS=OFF
+    - -D OpenGL_GL_PREFERENCE=GLVND
+
+  - name: appdata
+    sources:
+    - type: file
+      path: update_appdata.sh
+    buildsystem: simple
+    build-commands:
+    - ./update_appdata.sh /app/share/metainfo/org.mavlink.qgroundcontrol.metainfo.xml
```

```suggestion
  - name: QGroundControl
    buildsystem: cmake-ninja
    builddir: true
    config-opts:
    - -D BUILD_SHARED_LIBS=OFF
    - -D OpenGL_GL_PREFERENCE=GLVND
    post-install:
    - ./update_appdata.sh /app/share/metainfo/org.mavlink.qgroundcontrol.metainfo.xml
    sources:
    - type: git
      url: https://github.com/mavlink/qgroundcontrol.git
      commit: eca2614b6e2a237c3717f246c8c6e42d7d903aaf # Mar 1, 2023
    - type: file
      path: update_appdata.sh
```

But this should be a patch instead of a script.

---

I added a custom `org.mavlink.qgroundcontrol.appdata.xml` to the repo until the upstream patch (https://github.com/mavlink/qgroundcontrol/pull/10615) has been merged.

---

don't. Add a patch.

---

I don't understand. When I patched the AppData file, you said "you can NOT do that." (https://github.com/flathub/flathub/pull/3968#pullrequestreview-1335915434).

How shall I resolve this? Should I create my own `CC0-1.0` licensed AppData file, or patch the upstream AppData file to change their licence without permission?

---

"Changing the license" is not patching.

---

Does this mean I can create a patch that changes `<metadata_license>GPL-3.0 AND Apache-2.0</metadata_license>` to `<metadata_license>CC0-1.0</metadata_license>`?

How am I otherwise supposed to create an AppData file that passes flatpak builder?